src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1260,25 +1263,39 @@ private void checkExtraFields(int cenPos, int startingOffset,
                 zerror("Invalid CEN header (extra data field size too long)");
             }
             int currentOffset = startingOffset;
-            while (currentOffset < extraEndOffset) {
+            // Walk through each Extra Header. Each Extra Header Must consist of:
+            //       Header ID - 2 bytes
+            //       Data Size - 2 bytes:
+            while (currentOffset + Integer.BYTES <= extraEndOffset) {
                 int tag = get16(cen, currentOffset);
                 currentOffset += Short.BYTES;
 
                 int tagBlockSize = get16(cen, currentOffset);
+                currentOffset += Short.BYTES;
                 int tagBlockEndingOffset = currentOffset + tagBlockSize;
 
                 //  The ending offset for this tag block should not go past the
                 //  offset for the end of the extra field
                 if (tagBlockEndingOffset > extraEndOffset) {
-                    zerror("Invalid CEN header (invalid zip64 extra data field size)");
+                    // There are some extra headers with an invalid data size, if
+                    // found, we need to just ignore them
+                    break;
                 }
-                currentOffset += Short.BYTES;
 
                 if (tag == ZIP64_EXTID) {
                     // Get the compressed size;
                     long csize = CENSIZ(cen, cenPos);
                     // Get the uncompressed size;
                     long size = CENLEN(cen, cenPos);
+                    // if ZIP64_EXTID blocksize == 0 validate csize and size
+                    // to make sure neither field == ZIP64_MAGICVAL
+                    if (tagBlockSize == 0) {
+                        if ( csize == ZIP64_MAGICVAL || size == ZIP64_MAGICVAL) {
+                            zerror("Invalid CEN header (invalid zip64 extra data field size)");
+                        }
+                        // Only validate the ZIP64_EXTID data if the block size > 0
+                        return;
+                    }
                     checkZip64ExtraFieldValues(currentOffset, tagBlockSize,
```

Comment:
It seems in the spec there is no limitation on the tagBlockSize of the extra field, it only mention this:
```
      The following is the layout of the zip64 extended
      information "extra" block. If one of the size or
      offset fields in the Local or Central directory
      record is too small to hold the required data,
      a Zip64 extended information record is created.
      The order of the fields in the zip64 extended
      information record is fixed, but the fields MUST
      only appear if the corresponding Local or Central
      directory record field is set to 0xFFFF or 0xFFFFFFFF.

      Note: all fields stored in Intel low-byte/high-byte order.

        Value      Size       Description
        -----      ----       -----------
(ZIP64) 0x0001     2 bytes    Tag for this "extra" block type
        Size       2 bytes    Size of this "extra" block
        Original
        Size       8 bytes    Original uncompressed file size
```

What prevents it to be "61373"?